app/frontend/components/pages/delegations/delegatePage.js

```diff
@@ -0,0 +1,159 @@
+const {h} = require('preact')
+const connect = require('unistore/preact').connect
+const actions = require('../../../actions')
+const tooltip = require('../../common/tooltip')
+const printAda = require('../../../helpers/printAda')
+const {AdaIcon} = require('../../common/svg')
+
+const formatStakePoolInfo = (className, message) => {
+  return h('div', {class: `stake-pool-info ${className}`}, message)
+}
+
+const CalculatingFee = () => h('div', {class: 'validation-message send'}, 'Calculating fee...')
+
+const getStakePoolValidationMessage = (stakePool, stakePools) => {
+  let className, message
+
+  const addressUsages = stakePools.filter((pool) => pool.address === stakePool.address)
```

Comment:
What is `addressUsages` ?

Reply:
It contains the stake pools with the same ID (what I mistakenly called "address") as `stakePool` in the argument. This is here to check whether the user has typed a stake pool multiple times.